crates/config/src/config.rs

```diff
@@ -0,0 +1,282 @@
+    pub admin_enabled: bool,
+    /// The URI for the external Redis state store.
+    pub remote_state: Option<String>,
+    /// The number of trusted proxy hops expected to be exterior to Bulwark.
+    ///
+    /// This number does not include Bulwark or the proxy hosting it in the proxy hop count. Zero implies that
+    /// there are no other proxies exterior to Bulwark. This is used to ensure the `Forwarded` and `X-Forwarded-For`
+    /// headers are not spoofed. If this is set incorrectly, the client IP reported to plugins will be incorrect.
+    pub proxy_hops: u8,
+    // TODO: it may be useful to introduce an "auto" setting for `proxy_hops` since it's possible to auto-discover
+}
+
+/// The default [`Service::port`] value.
+pub const DEFAULT_PORT: u16 = 8089;
+/// The default [`Service::admin_port`] value.
+pub const DEFAULT_ADMIN_PORT: u16 = 8090;
+
+/// Configuration for the decision thresholds.
+///
+/// No threshold is necessary for the default `allowed` outcome because it is defined by the range between the
+/// `suspicious` threshold and the `trusted` threshold. The thresholds must have values in descending order, with
+/// `restrict` > `suspicious` > `trusted`. None of the threshold values may be equal.
+#[derive(Debug, Clone, Copy)]
+pub struct Thresholds {
+    /// True if the primary service should take no action in response to restrict decisions.
+    pub observe_only: bool,
+    /// Any decision value above the `restrict` threshold will cause the corresponding request to be blocked.
+    pub restrict: f64,
+    /// Any decision value above the `suspicious` threshold will cause the corresponding request to be flagged as
+    /// suspicious but it will still be allowed.
+    pub suspicious: f64,
+    /// Any decision value below the `trust` threshold will cause the corresponding request to be flagged as trusted.
+    /// This primarily affects plugins which use feedback loops.
+    pub trust: f64,
+}
+
+/// The default [`Thresholds::observe_only`] value.
+pub const DEFAULT_OBSERVE_ONLY: bool = false;
+/// The default [`Thresholds::restrict`] value.
+pub const DEFAULT_RESTRICT_THRESHOLD: f64 = 0.8;
+/// The default [`Thresholds::suspicious`] value.
+pub const DEFAULT_SUSPICIOUS_THRESHOLD: f64 = 0.6;
+/// The default [`Thresholds::trust`] value.
+pub const DEFAULT_TRUST_THRESHOLD: f64 = 0.2;
+
+impl Default for Thresholds {
+    /// Default decision thresholds.
+    fn default() -> Self {
+        Self {
+            observe_only: DEFAULT_OBSERVE_ONLY,
+            restrict: DEFAULT_RESTRICT_THRESHOLD,
+            suspicious: DEFAULT_SUSPICIOUS_THRESHOLD,
+            trust: DEFAULT_TRUST_THRESHOLD,
+        }
+    }
+}
+
+/// The configuration for an individual plugin.
+///
+/// This structure will be wrapped by structs in the host environment.
+#[derive(Debug, Validate, Clone, Default)]
+pub struct Plugin {
+    /// The plugin reference key. Should be limited to ASCII lowercase a-z plus underscores.
```

Comment:
It has to be handled elsewhere since there's no way to know from within a `Plugin` struct what other `Plugin` structs exist. That needs to be addressed in the function that resolves references.